HLTrigger/Configuration/python/customizeHLTforCMSSW.py

```diff
@@ -170,10 +170,27 @@ def customiseFor2017DtUnpacking(process):
 
     return process
 
+
+# Update Pixel Gain calibration scheme for Run3 (PR #29333)
+def customiseFor29333(process):
+    for producer in producers_by_type(process, "SiPixelClusterProducer"):
+        # For Run3, change in the gain calibration scheme,
+        # to include the VCal conversion factors directly in the SiPixelGainCalibration DB object
+        # full details at: https://indico.cern.ch/event/879470/contributions/3796405/attachments/2009273/3356603/pix_off_25_3_gain_calibration_mc.pdf
+        from Configuration.Eras.Modifier_run3_common_cff import run3_common
+        run3_common.toModify(producer,
+                             VCaltoElectronGain      = 1,  # all gains=1, pedestals=0
```

Comment:
Could please you code this without resorting to eras? Ie, direct modification of the ```prodcuer``` parametrs.

Reply:
@Martin-Grunewald why? We don't want to change Run2 ...

Reply:
These changes will be imported in the HLT menus in ConfDB, with menu dumps redone, and then removed from the customisation. HLT does not use eras, and the menu dumps conform to Run3.

Reply:
but won't this change the HLT results for Run2  as well, in the meanwhile before the menus are re-dumped from ConfDB? I am at loss here. Anyway, just for me to understand you want directly:
```diff
diff --git a/HLTrigger/Configuration/python/customizeHLTforCMSSW.py b/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
index 5239d360cf2..6109d32903f 100644
--- a/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
+++ b/HLTrigger/Configuration/python/customizeHLTforCMSSW.py
@@ -177,13 +177,11 @@ def customiseFor29333(process):
         # For Run3, change in the gain calibration scheme,
         # to include the VCal conversion factors directly in the SiPixelGainCalibration DB object
         # full details at: https://indico.cern.ch/event/879470/contributions/3796405/attachments/2009273/3356603/pix_off_25_3_gain_calibration_mc.pdf
-        from Configuration.Eras.Modifier_run3_common_cff import run3_common
-        run3_common.toModify(producer,
-                             VCaltoElectronGain      = 1,  # all gains=1, pedestals=0
-                             VCaltoElectronGain_L1   = 1,
-                             VCaltoElectronOffset    = 0,
-                             VCaltoElectronOffset_L1 = 0
-                             )
+        producer.VCaltoElectronGain      = cms.int32(1) # all gains=1, pedestals=0
+        producer.VCaltoElectronGain_L1   = cms.int32(1)
+        producer.VCaltoElectronOffset    = cms.int32(0)
+        producer.VCaltoElectronOffset_L1 = cms.int32(0)
+
     return process
 
 # CMSSW version specific customizations
```
?

Reply:
Yes, indeed. The HLT menu in th erelease is for Run3, we can not keep HLT menus for previous running periods in th erelease, so run2 workflows are using the fake2 menu.